rest_framework/renderers.py

```diff
@@ -604,10 +604,13 @@ def get_description(self, view, status_code):
     def get_breadcrumbs(self, request):
         return get_breadcrumbs(request.path, request)
 
-    def get_extra_actions(self, view):
-        if hasattr(view, 'get_extra_action_url_map'):
-            return view.get_extra_action_url_map()
-        return None
+    def get_extra_actions(self, view, status_code):
+        if (status_code in (status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN)):
```

Comment:
Is this handling the case when only some extra actions require auth? I was thinking of taking this but we would need to update `get_extra_action_url_map` to keep auth in mind and only return actions where user has permission.

Reply:
> Is this handling the case when only some extra actions require auth?

No, this isn't about per-view auth/perms checking and constructing a list from that. I don't think we'd want to go that route, as this require simulating a new request per-view, which would be relatively expensive. 

This is just about checking whether the user is authenticated for the current page. If a user isn't authenticated, the page doesn't display the serializer, filters, etc.. however, the list of actions currently does render, which is inconsistent.